### DISCUSSION TITLE
How should we serialize highlights and bookmarks?

### DISCUSSION THREAD

**Discussion by llemeurfr** [readium/annotations#12] [General]
Highlights and bookmarks are annotations without body. 

How should we express it in the spec, so that we can serialize both?

**Comment by llemeurfr** [comment]
In the W3C Annotation Model I see "Annotations have 0 or more Bodies."

-> seems that this is the best answer. A highlight / bookmark is serialized as an annotation with no body. 

For a bookmark, the target selector is a "pointer" to a precise location in the content, not a range selector (segment).

There must be a semantic way to disambiguate annotations / highlights and bookmarks. The `type` of the JSON-LD structure seems the best choice. We'll add a "Bookmark" value as _annotation_ type

**Comment by HadrienGardeur** [comment]
There's a very thin line between:

- bookmarks
- highlights
- and annotations

Bookmarks usually target:

- a page in a pre-paginated publication (Divina, FXL or PDF)
- a timestamp in an audiobook
- and a position in a reflowable EPUB

In some cases, bookmarks may also have accompanying tag(s) and a note from the user, but that's a more advanced use case than what's available in most apps.

Highlights and annotations usually target:

- a text fragment in textual publications
- an image or a fragment of an image in visual publications

The concept of highlights/annotations doesn't really exist in audiobooks.

Highlights and annotations usually have a color as well, and may have variations in styles supported (for example underlined). They may also offer tags.

The main difference that differentiates an annotation from a highlight being the presence of an accompanying note.

**Comment by llemeurfr** [comment]
Note: in https://www.w3.org/TR/annotation-model/#cardinality-of-bodies-and-targets, the model states that bookmarks and highlights are simply annotations with no body.

**Comment by llemeurfr** [comment]
To identify bookmarks without creation a new type ("type": "Bookmark") I've found that the W3C Annotation model defines a specific "motivation": "bookmarking" property. 
I've added it to our profile.
